### PR TITLE
fix jsonrpsee: feature renamed `client` -> `async-client`

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

File: jsonrpsee/Cargo.toml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jsonrpsee-types = { path = "../types", version = "0.6.1", package = "jsonrpsee-t
 [features]
 client-ws-transport = ["jsonrpsee-client-transport/ws", "jsonrpsee-client-transport/tls"]
 client-ws-transport-no-tls = ["jsonrpsee-client-transport/ws"]
-core-client = ["jsonrpsee-core/client"]
+core-client = ["jsonrpsee-core/async-client"]
```
Comment (dvdplm) [inline jsonrpsee/Cargo.toml:27]: Maybe `async-client` is a better name for the feature, so it's consistent and we avoid the ambiguous "core"?

Comment (niklasad1) [reply jsonrpsee/Cargo.toml:27]: agree.

File: jsonrpsee/src/lib.rs [hunk 2]
```diff
@@ -79,7 +82,13 @@ pub use jsonrpsee_types as types;
 #[cfg(any(feature = "http-server", feature = "ws-server"))]
 pub use jsonrpsee_core::server::rpc_module::{RpcModule, SubscriptionSink};
 
-#[cfg(any(feature = "http-server", feature = "ws-server", feature = "core-client"))]
+#[cfg(any(
+	feature = "http-server",
+	feature = "ws-server",
+	feature = "async-client",
```
Comment (dvdplm) [inline jsonrpsee/src/lib.rs:88]: do we need `feature = "client"` as well?

**Review by dvdplm — APPROVED** [review]

File: jsonrpsee/src/lib.rs (hunk 2 above)
Comment (niklasad1) [reply jsonrpsee/src/lib.rs:89]: I don't think so but added it too the be on the safe side :)

**Pull request merged**